src/Microsoft.DotNet.ProjectModel/Graph/BuildAction.cs

```diff
@@ -0,0 +1,81 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.DotNet.ProjectModel.Graph
+{
+    public struct BuildAction : IEquatable<BuildAction>
+    {
+        public static readonly BuildAction Compile = new BuildAction(nameof(Compile));
+        public static readonly BuildAction EmbeddedResource = new BuildAction(nameof(EmbeddedResource));
+        public static readonly BuildAction Resource = new BuildAction(nameof(Resource));
+
+        // Default value
+        public static readonly BuildAction None = new BuildAction(nameof(None));
+
+        public string Value { get; }
+
+        private BuildAction(string value)
+        {
+            Value = value;
+        }
+
+        public static bool TryParse(string value, out BuildAction type)
+        {
+            // We only support values we know about
+            if (string.Equals(Compile.Value, value, StringComparison.OrdinalIgnoreCase))
+            {
+                type = Compile;
+                return true;
+            }
+            else if (string.Equals(EmbeddedResource.Value, value, StringComparison.OrdinalIgnoreCase))
+            {
+                type = EmbeddedResource;
+                return true;
+            }
+            else if (string.Equals(Resource.Value, value, StringComparison.OrdinalIgnoreCase))
+            {
+                type = Resource;
+                return true;
+            }
+            type = None;
+            return false;
+        }
+
+        public override string ToString()
+        {
+            return Value;
```

Comment:
I would love to have it changed as well. But one at a time. `ToString` is used for debugging in my opinion it's clearer to have type of the object in the print.

Non-block.